crates/noirc_evaluator/src/ssa/mem.rs

```diff
@@ -23,6 +23,41 @@ impl ArrayId {
     pub fn dummy() -> ArrayId {
         ArrayId(std::u32::MAX)
     }
+    // TODO: docs
+    pub fn get_dummy_store(&self, context: &SsaContext) -> NodeId {
+        context.dummy_store[self]
+    }
+
+    // TODO: docs
+    pub fn get_dummy_load(&self, context: &SsaContext) -> NodeId {
+        context.dummy_load[self]
+    }
+
+    // TODO: docs
```

Comment:
Should we create an issue for documenting the dummy store/load scheme as a whole? I think the whole system could benefit from an explanation of what it is, why it exists, etc.

Reply:
Might be best to do it here in this PR -- easiest way for me is to try and document everything I touch